### PR TITLE
Body validation

### PR DESCRIPTION
**Pull request by delvedor**
As titled, discussion in #12 .

Few links:
- [ajv](https://github.com/epoberezkin/ajv)
  - [options](https://github.com/epoberezkin/ajv#options)
  - [coercion](https://github.com/epoberezkin/ajv/blob/master/COERCION.md)

### CODE REVIEW

File: fastify.js [hunk 1]
```diff
@@ -1,11 +1,45 @@
...
 const wayfarer = require('wayfarer')
+const stripUrl = require('pathname-match')
+const urlUtil = require('url')
 const fastJsonStringify = require('fast-json-stringify')
 const fastSafeStringify = require('fast-safe-stringify')
+const Ajv = require('ajv')
 const jsonParser = require('body/json')
-const schema = Symbol('schema')
+
+const inSchema = Symbol('inSchema')
```
Comment (mcollina) [inline fastify.js]: this should be `payloadSchema`, we will need another schema for querystring.

File: test/put.test.js [hunk 2]
```diff
@@ -1,3 +1,5 @@
 'use strict'
 
-require('./helper').payloadMethod('put', require('tap'))
+const t = require('tap')
+require('./helper').payloadMethod('put', t)
```
Comment (mcollina) [inline test/put.test.js]: we should probably rename `'helpder'`into something more meaningful?

File: fastify.js [hunk 3]
```diff
@@ -1,11 +1,45 @@
...
+            type: 'string'
+          }
+        }
+      },
+      message: {
+        type: 'string'
+      }
+    }
+  }
+})
```
Comment (mcollina) [inline fastify.js]: we can probably move this into `lib/`, this file is starting to get big.

File: fastify.js [hunk 4]
```diff
@@ -93,7 +131,7 @@ function build () {
       const handle = node[req.method]
 
       if (handle && req.method === 'GET') {
-        handleNode(handle, params, req, res, null)
+        handleNode(handle, params, req, res, urlUtil.parse(req.url, true).query)
```
Comment (mcollina) [inline fastify.js]: querystrings are not body!

Comment (delvedor) [inline fastify.js]: I know, but since this pr is not trivial, for the moment I'm using the same variable to get it work.
In your opinion which is the best way to handle this?

Comment (mcollina) [inline fastify.js]: We can validate querystrings _before_ calling `handleNode`, and before parsing the body. At the end, the querystring needs to go into `Request.query`.

Comment (delvedor) [inline fastify.js]: I prefer the second option, considering that managing querystring is not trivial, probably is better do a different pr for that, and in this one focus on the body validation.

File: test/put.test.js (hunk 2 above)
Comment (delvedor) [inline test/put.test.js]: Agreed, but not in this pr :)
